dsmr/block.go

```diff
@@ -111,38 +111,60 @@ func parseChunkProto[T Tx](chunkProto *dsmr.Chunk) (Chunk[T], error) {
 type ChunkFault struct{}
 
 type Block struct {
-	ID        ids.ID `serialize:"true"`
```

Comment:
We cannot include the block's own ID in the serialization. The blkID is a hash of the block bytes, so including the ID in the block bytes creates a cycle.